crates/ide-assists/src/handlers/convert_let_else_to_match.rs

```diff
@@ -0,0 +1,441 @@
+use syntax::ast::{edit::AstNodeEdit, AstNode, HasName, LetStmt, Pat};
+use syntax::T;
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+/// Gets a list of binders in a pattern, and whether they are mut.
+fn binders_in_pat(pat: &Pat) -> Option<Vec<(String, bool)>> {
```

Comment:
Can we change this to `fn binders_in_pat(acc: &mut Vec<(ast::Name,bool)>, pat: &Pat) -> Option<()>`? That way we don't allocate a bunch for some patterns